src/SingleVault.sol

```diff
@@ -3,29 +3,22 @@ pragma solidity ^0.8.24;
 
 import {
     ERC4626Upgradeable,
-    AccessControlUpgradeable,
     ReentrancyGuardUpgradeable,
     TimelockControllerUpgradeable,
-    IERC20
+    IERC20,
+    IERC4626,
+    Math
 } from "src/Common.sol";
 
 import {ISingleVault} from "src/ISingleVault.sol";
+import {SlisBnbModule} from "src/module/SlisBnbModule.sol";
```

Comment:
This should be functionally equivalent to having the code in contract as before.

The library is deployed external so it doesn't increase the bytecode of the contract, and in theory can be called by other contracts

But must be noted to upgrade the logic in the module we still need to upgrade the whole implementation of the SingleVault.

Also because of how libraries work in solidity, there's no way to have a library interface so it makes this contract import the concrete library so SingleVault can't be generic (as opposed to it depending on a generic IBnbModule contract)

Reply:
Yeah, deploying a new SingleVault which has an updated SlisBnbModule will require redeploying the updated SlisBnbModule. But this should all be handled by foundry once you deploy the new SingleVault implementation (which imports the new SlisBnbModule), after which you upgrade the proxy to point to this new impl.

Reply:
Yes, agree. Thanks for this review. I've reverted efforts to make the SingleVault generic and will just deploy the basic functionality needed for ynBNB now.